CMakeLists.txt

```diff
@@ -32,28 +36,35 @@ ENDIF()
 # Activate this to error on all things deprecated in 5.15
 #add_compile_definitions("QT_DISABLE_DEPRECATED_BEFORE=0x050F00")
 
-# If Qt6, set series number to 1.
-IF(Qt6_FOUND)
-     SET(STELLARIUM_MAJOR 1)
+# two last digits of the current year
+STRING(TIMESTAMP SHORTYEAR "%y")
+MATH(EXPR STELLARIUM_MAJOR_DIGIT "${SHORTYEAR} + 0" OUTPUT_FORMAT DECIMAL)
+# Version of Stellarium has format YY.V.YD, where:
+# -- YY (MAJOR) - two last digits of the year of release
+# -- V  (MINOR) - version of the release (0 is used before first release)
+# -- YD (PATCH) - 0 for releases and the day of the current year for snapshots
+# Example: first release in year 2023 has version 23.1.0 and short (public) version 23.1, series 23.0
+# Note: Use integer versions instead of strings for easier handling if required
+SET(STELLARIUM_MAJOR ${STELLARIUM_MAJOR_DIGIT})
+SET(STELLARIUM_MINOR ${STELLARIUM_RELEASE_NUMBER})
+IF(STELLARIUM_RELEASE_BUILD)
+    ADD_DEFINITIONS(-DSTELLARIUM_RELEASE_BUILD)
+    SET(STELLARIUM_PATCH 0)
 ELSE()
-     SET(STELLARIUM_MAJOR 0)
+    ########### Get revision number for non-release builds ###########
+    STRING(TIMESTAMP CDNUMBER "%j")
+    MATH(EXPR STELLARIUM_PATCH_DIGIT "${CDNUMBER} + 0" OUTPUT_FORMAT DECIMAL)
+    SET(STELLARIUM_PATCH ${STELLARIUM_PATCH_DIGIT})
 ENDIF()
 # Use integer versions instead of strings for easier handling if required
-SET(STELLARIUM_MINOR 22)
-SET(STELLARIUM_PATCH 5)
 ADD_DEFINITIONS(
      -DSTELLARIUM_MAJOR=${STELLARIUM_MAJOR}
      -DSTELLARIUM_MINOR=${STELLARIUM_MINOR}
      -DSTELLARIUM_PATCH=${STELLARIUM_PATCH}
 )
 SET(VERSION "${STELLARIUM_MAJOR}.${STELLARIUM_MINOR}.${STELLARIUM_PATCH}")
-# Public version has format YY.V, where:
-# -- YY - two last digits of the year of release
-# -- V  - version of the release
-# Example: first release in year 2023 has version 1.23.0 and public version 23.1
-MATH(EXPR STELLARIUM_V "${STELLARIUM_PATCH} + 1" OUTPUT_FORMAT DECIMAL)
```

Comment:
Yes, this was truly confusing. {0|1}.23.0 or {0|1}.23.1 would have been consistent "internal" numbers for 23.0 or 23.1, respectively.

Reply:
I've omitted first {0|1}, because it's not critical.

Reply:
It was a signal to discern Qt5- and Qt6-based builds. Is this now done in the package filename?

Reply:
> It was a signal to discern Qt5- and Qt6-based builds. Is this now done in the package filename?

Yes for Windows and macOS packages